client/lib/fifo/fifo_windows.go

```diff
@@ -0,0 +1,69 @@
+package fifo
+
+import (
+	"io"
+	"net"
+
+	winio "github.com/Microsoft/go-winio"
+)
+
+const PipeBufferSize = int(^uint16(0))
+
+type FIFO struct {
+	listener net.Listener
+	conn     net.Conn
+}
+
+func (f *FIFO) Read(p []byte) (n int, err error) {
+	if f.conn != nil {
+		c, err := f.listener.Accept()
+		if err != nil {
+			return 0, err
+		}
+
+		f.conn = c
+	}
+	return f.conn.Read(p)
+}
+
+func (f *FIFO) Write(p []byte) (n int, err error) {
+	if f.conn != nil {
+		c, err := f.listener.Accept()
+		if err != nil {
+			return 0, err
+		}
+
+		f.conn = c
+	}
+	return f.conn.Write(p)
+}
+
+func (f *FIFO) Close() error {
+	if f.conn != nil {
```

Comment:
`Conn.Close` is threadsafe and `File.Close` not being threadsafe was considered a bug: https://github.com/golang/go/issues/7970

Perhaps we should lock around the `conn` check? The only way to unblock a blocked Read() or Write() is to Close it.